jdaviz/configs/cubeviz/plugins/viewers.py

```diff
@@ -167,6 +191,10 @@ def active_image_layer(self):
 
         return visible_layers[-1]
 
+    def _on_global_display_unit_changed(self, msg):
+        del self.slice_values
+        self.slice_values
```

Comment:
This would really benefit from #2253, but for now, should be:
```suggestion
        if 'slice_values' in self.__dict__:
            del self.__dict__['slice_values']
```

(and same for the other case)